server/game/gameSystems/LookAtSystem.ts

```diff
@@ -0,0 +1,80 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { BaseCard } from '../core/card/BaseCard';
+import { EventName } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+
+export interface ILookAtProperties extends ICardTargetSystemProperties {
+    message?: string | ((context) => string);
+    messageArgs?: (cards: any) => any[];
+}
+
+export class LookAtSystem extends CardTargetSystem<ILookAtProperties> {
+    public override readonly name = 'lookAt';
+    public override readonly eventName = EventName.OnLookAtCards;
+    public override readonly effectDescription = 'look at a card';
+
+    protected override defaultProperties: ILookAtProperties = {
+        message: '{0} sees {1}'
+    };
+
+    public override canAffect(card: BaseCard, context: AbilityContext) {
+        // TODO: What situations would mean that a card cannot be looked at?
+        // if (!card.isFacedown() && (card.isInProvince() || card.location === Locations.PlayArea)) {
+        //     return false;
+        // }
+        return super.canAffect(card, context);
+    }
+
+    public override generateEventsForAllTargets(context: AbilityContext, additionalProperties = {}): GameEvent[] {
+        const events: GameEvent[] = [];
+
+        const { target } = this.generatePropertiesFromContext(context, additionalProperties);
+        const cards = (target as BaseCard[]).filter((card) => this.canAffect(card, context));
+        if (cards.length === 0) {
+            return [];
+        }
+        const event = this.createEvent(null, context, additionalProperties);
+        this.updateEvent(event, cards, context, additionalProperties);
+        events.push(event);
+
+        return events;
+    }
+
+    public override addPropertiesToEvent(event, cards, context: AbilityContext, additionalProperties): void {
+        if (!cards) {
+            cards = this.generatePropertiesFromContext(context, additionalProperties).target;
+        }
+        if (!Array.isArray(cards)) {
+            cards = [cards];
+        }
+        event.cards = cards;
+        const obj = { a: cards, b: context };
+        event.stateBeforeResolution = cards.map((a) => {
```

Comment:
Looks like this is only here because a couple of L5R cards had a specific kind of reaction to `LookAt.` Fine to leave it here or remove it, but if we leave it here we should add a note as to what it's intended for